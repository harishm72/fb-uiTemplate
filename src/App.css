.App {
  margin: auto;
  padding: 1em;
  width: 25em;
  border: 2px solid black;
}
.App select, label{
  margin-left: 1em;
}
.post{
  display: flex;
  flex-direction: column;
  padding-bottom: 1em;
}
.post-type{
  display: flex;
  flex-direction: column;
}
.like-button:focus{
  background-color: black;
  color: white;
}
.comment{
  border: 1px solid black;
  padding-bottom: 0.5em;
}