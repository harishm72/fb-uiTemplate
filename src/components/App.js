import React, { Component } from 'react';
import Header from './Header';
import NewsFeed from './NewsFeed';

class App extends Component {

  constructor(props){
    super(props)
    this.state = {
      title : "facebook",
      OptionSelected : "imageAndText",
      data: []
    }
  }

  componentDidMount(){
    this.setState({data : [
      {
      "id" : 1,
      "isLiked" : false,
      "item_description": "Declare variables not wars!!",
      "image": "",
      "likes": 10,
      "comments": [{
          "comment": "Well said, i agree.",
          "created_at": "Fri May 11 2018 19:17:43 GMT+0530 (IST)"
        },
        {
          "comment": "Good one!",
          "created_at": "Fri May 10 2018 15:23:36 GMT+0530 (IST)"
        }
      ]
    },
    {
      "id": 2,
      "isLiked" : false,
      "item_description": "This is an item with an image",
      "image": "./images/post1.jpg",
      "likes": 20,
      "comments": [{
          "comment": "Spectacular view",
          "created_at": "Fri May 11 2018 19:17:43 GMT+0530 (IST)"
        },
        {
          "comment": "Vacation...?",
          "created_at": "Fri May 10 2018 15:23:36 GMT+0530 (IST)"
        }
      ]
    }
  ] })
  }
  ModifyContent = (event) => {
    let value = event.target.value 
    this.setState({ OptionSelected : value })
  }
  HandleLike = (id) => {
    this.setState({
      data: this.state.data.map(post => {
        if (post.id === id) {
          let newPost = {...post}
          newPost.isLiked = !newPost.isLiked;
          newPost.likes = newPost.isLiked ? newPost.likes + 1 : newPost.likes - 1
          return newPost;
        }
        return post;
      })
    })
  }

  HandleComment = (newComment, id) => {
    this.setState({
      data: this.state.data.map(post => {
        if (post.id === id) {
          post.comments.push({
            "comment": newComment[0],
            'created_at': newComment[1]
          })
        }
        return post
      })
    })
  }
  
  render() {
    let option = this.state.OptionSelected
        let feedItems = [];
        switch(option){
            case "image":
            feedItems = this.state.data.filter(post => post['image'])
            break;

            case "text":
            feedItems = this.state.data.filter(post => !post['image']);
            break;

            case "none":
            feedItems = [];
            break;

            default:
            feedItems = this.state.data
            break;
        }

    if(feedItems){
    return (
      <div>
        <div className="app-header">
        <Header title={this.state.title}/>
        </div>
     
      <div className="app">
      <label>
        Select-type
        <select defaultValue="imageAndText"  onChange={this.ModifyContent} className="post-type">
          <option value="text" >Text Only</option>
          <option value = "image" >Image Only</option>
          <option  value ="imageAndText" >Image and Text</option>
          <option value="none" >No items</option>
        </select>
        </label>
        <NewsFeed feed={feedItems} HandleLike={this.HandleLike} HandleComment={this.HandleComment}/>
      </div>
      </div>
    );
    }
    else return(<p>loading....</p>)
  }
  
}



export default App;
