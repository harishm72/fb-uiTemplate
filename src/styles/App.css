img{
  max-width: 37.75em;
}
body{
  background-color: #e9ebee;
}
.app-header{
  background-color: #4267b2;
  margin: 0 !important;
  height: 54px;
}
h1{
  margin: 0;
}
.app {
  margin: 2em auto;
  padding: 1em;
  width: 40em;
  border: 2px solid black;
  border-radius: 4px;
  background-color: azure;
}
.app select, label{
  margin-left: 1em;
}
.post{
  display: flex;
  flex-direction: column;
  padding-bottom: 1em;
}
.post-type{
  display: flex;
  flex-direction: column;
}
.like-button:focus{
  background-color: black;
  color: white;
}
.comment{
  border: 1px solid black;
  padding-bottom: 0.5em;
}
.new-comment{
  height: 2em;
}
.comment-input{
  width: 99.5%;
  height: 1.65em;
  font-size: inherit;
}
.post-item{
  border-radius: 4px;
  border: 1px solid black;
  margin: 1em;
}