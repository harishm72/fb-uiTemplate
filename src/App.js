import React, { Component } from 'react';
import './App.css';
import Post from './components/Post';

class App extends Component {

  constructor(props){
    super(props)
    this.state = {
      OptionSelected : "imageAndText",
      data: [{
        "item_description": "This is a text only item",
        "image": "",
        "likes": 10,
        "comments": [{
            "comment": "Well said, i agree.",
            "created_at": "Fri May 11 2018 19:17:43 GMT+0530 (IST)"
          },
          {
            "comment": "Good one!",
            "created_at": "Fri May 10 2018 15:23:36 GMT+0530 (IST)"
          }
        ]
      },
      {
        "item_description": "This is an item with an image",
        "image": "/path/to/image.png",
        "likes": 20,
        "comments": [{
            "comment": "Spectacular view",
            "created_at": "Fri May 11 2018 19:17:43 GMT+0530 (IST)"
          },
          {
            "comment": "Vacation...?",
            "created_at": "Fri May 10 2018 15:23:36 GMT+0530 (IST)"
          }
        ]
      }
    ]
    }
  }

  ModifyContent = (event) => {
    let value = event.target.value 
    this.setState(prev => ({
      OptionSelected : value
    }))
  }
  render() {
    let option = this.state.OptionSelected
        let posts = [];
        switch(option){
            case "image":
            posts = this.state.data.filter(post => post['image'])
            break;

            case "text":
            posts = this.state.data.filter(post => !post['image']);
            break;

            case "none":
            posts = [];
            break;

            default:
            posts = this.state.data
            break;
        }

    return (
      <div className="App">
      <label>
        Select-type
        <select defaultValue="imageAndText"  onChange={this.ModifyContent} className="post-type">
          <option value="text" >Text Only</option>
          <option value = "image" >Image Only</option>
          <option  value ="imageAndText" >Image and Text</option>
          <option value="none" >No items</option>
        </select>
        </label>
        <Post post={posts}/>
      </div>
    );
  }
}
export default App;
