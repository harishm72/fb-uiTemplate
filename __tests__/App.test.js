import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../src/components/App';
import renderer from 'react-test-renderer'

describe('>>>H O M E --- Snapshot',()=>{
  it('Rednders the App Component', () => {
      const renderedValue =  renderer.create(<App />).toJSON()
      expect(renderedValue).toMatchSnapshot();
  });
});

describe('>>>H O M E --- Shallow Render REACT COMPONENTS',()=>{
  let wrapper

  beforeEach(()=>{
      wrapper = shallow(<App/>) 
  })

  it('+++ render the DUMB component', () => {
     expect(wrapper.length).toEqual(1)
  });
  
});