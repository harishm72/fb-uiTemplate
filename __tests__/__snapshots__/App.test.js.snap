exports[`>>>H O M E --- Snapshot Rednders the App Component 1`] = `
<div>
  <div
    className="app-header">
    <div>
      <h1>
        facebook
      </h1>
    </div>
  </div>
  <div
    className="app">
    <label>
      Select-type
      <select
        className="post-type"
        defaultValue="imageAndText"
        onChange={[Function]}>
        <option
          value="text">
          Text Only
        </option>
        <option
          value="image">
          Image Only
        </option>
        <option
          value="imageAndText">
          Image and Text
        </option>
        <option
          value="none">
          No items
        </option>
      </select>
    </label>
    <div>
      <div
        className="post-item">
        <p>
          Declare variables not wars!!
        </p>
        <div>
          <p>
            10
             Likes
          </p>
          <button
            className="like-button"
            onClick={[Function]}>
            Like
          </button>
        </div>
        <div>
          <h3>
            Comments..
          </h3>
          <div
            className="comment">
            <p>
              Well said, i agree.
            </p>
            <p
              className="comment-time">
              Created at : 
              Fri May 11 2018 19:17:43 GMT+0530 (IST)
            </p>
          </div>
          <div
            className="comment">
            <p>
              Good one!
            </p>
            <p
              className="comment-time">
              Created at : 
              Fri May 10 2018 15:23:36 GMT+0530 (IST)
            </p>
          </div>
          <div
            className="new-comment">
            <form
              onSubmit={[Function]}>
              <input
                className="comment-input"
                onChange={[Function]}
                placeholder="type a comment here...."
                value="" />
            </form>
          </div>
        </div>
      </div>
      <div
        className="post-item">
        <p>
          This is an item with an image
        </p>
        <img
          alt=""
          src="./images/post1.jpg" />
        <div>
          <p>
            20
             Likes
          </p>
          <button
            className="like-button"
            onClick={[Function]}>
            Like
          </button>
        </div>
        <div>
          <h3>
            Comments..
          </h3>
          <div
            className="comment">
            <p>
              Spectacular view
            </p>
            <p
              className="comment-time">
              Created at : 
              Fri May 11 2018 19:17:43 GMT+0530 (IST)
            </p>
          </div>
          <div
            className="comment">
            <p>
              Vacation...?
            </p>
            <p
              className="comment-time">
              Created at : 
              Fri May 10 2018 15:23:36 GMT+0530 (IST)
            </p>
          </div>
          <div
            className="new-comment">
            <form
              onSubmit={[Function]}>
              <input
                className="comment-input"
                onChange={[Function]}
                placeholder="type a comment here...."
                value="" />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
